client/state/login/actions.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Internal dependencies
+ */
+import {
+	LOGIN_REQUEST,
+	LOGIN_REQUEST_FAILURE,
+	LOGIN_REQUEST_SUCCESS,
+} from 'state/action-types';
+import wp from 'lib/wp';
+import { externalRedirect } from 'lib/route/path';
+
+/**
+ * Attempt to login a user.
+ *
+ * @param  {String}    username_or_email  Username or email of the user.
+ * @param  {String}    password           Password of the user.
+ * @return {Function}                     Action thunk to trigger the login process.
+ */
+export const loginUser = ( username_or_email, password ) => {
+	return ( dispatch ) => {
+		dispatch( {
+			type: LOGIN_REQUEST,
+			username_or_email,
+			password
+		} );
+
+		return wp.undocumented().verifyLogin( username_or_email, password )
+			.then( ( data ) => {
+				dispatch( {
+					type: LOGIN_REQUEST_SUCCESS,
+					username_or_email,
+					password,
+					data,
+				} );
+			} ).catch( ( error ) => {
+				dispatch( {
+					type: LOGIN_REQUEST_FAILURE,
+					username_or_email,
+					password,
+					error: error.message
+				} );
+			} );
+	};
+};
+
+export const redirect = ( url ) => {
```

Comment:
This isn't a Redux action, so I don't think we should include it here. We can simply call `externalRedirect` in `Login` like we call `page` without passing it through a `connect` prop.